### PR TITLE
feat: user auth flow

### PR DESCRIPTION
**Pull request by morenocantoj**
This PR adds two new endpoints to register and authenticate an user against the platform:

- `POST /register` takes three mandatory parameters `email`, `nickname` and `password` and creates an `User`
- `POST /login` takes two mandatory parameters `email` and `password` and returns an `auth_token` that client could use to be authenticated against protected endpoints

To achieve so, a new `User` class has been added, along with an `UserModel` that represents an `User` in its `SQLAlchemy` implementation. New table `users` has been added to the database too.

Some restrictions to user auth flow have been applied:

- `nickname` and `email` are unique. This restriction applies to both attributes indifferently from the other
-  Passwords are hashed with [Argon2](https://pypi.org/project/argon2-cffi/) before being inserted into database
- Auth tokens are generated by [JWT](https://jwt.io/introduction) using `HS256` as the digital signature algorithm. The plan is to migrate to `RS256` in a future, which lets us using a private and a public certificate to encode/decode auth tokens

### CODE REVIEW

File: src/api/shared/handle_http_exception.py [hunk 1]
```diff
@@ -16,10 +18,49 @@ def handle_exception(error):
...
+        ),
+        DOMAIN_CATEGORY_EXCEPTIONS_TO_HTTP[error.category],
+    )
+
+
+@errors_bp.app_errorhandler(Exception)
+def handle_uknown_exception(*args):
+    """Handle unknown errors and return them in a JSON format"""
+
+    print(*args)
```
Comment (morenocantoj) [inline src/api/shared/handle_http_exception.py]: Remove this print

File: src/core/recipes/infrastructure/recipe_model.py [hunk 2]
```diff
@@ -42,11 +42,19 @@ def to_domain_object(self) -> Recipe:
...
+                "cover": self.cover,
             }
+            # {
+            #     "id_": cast(self.id, Integer),
+            #     "uuid": cast(self.uuid, String),
+            #     "name": cast(self.name, String),
+            #     "favourite_amount": cast(self.favourite_amount, Integer),
+            #     "preparation_time": cast(self.preparation_time, Integer),
+            #     "cover": cast(self.cover, String),
+            # }
```
Comment (morenocantoj) [inline src/core/recipes/infrastructure/recipe_model.py]: Remove this comment

File: src/api/shared/handle_http_exception.py (hunk 1 above)
Comment (morenocantoj) [reply src/api/shared/handle_http_exception.py]:
```suggestion
```

File: src/core/recipes/infrastructure/recipe_model.py (hunk 2 above)
Comment (morenocantoj) [reply src/core/recipes/infrastructure/recipe_model.py]:
```suggestion
```

File: src/api/shared/handle_http_exception.py (hunk 1 above)
Comment (morenocantoj) [inline src/api/shared/handle_http_exception.py]:
```suggestion
```

**Review by jorgebodega — APPROVED** [review]

**Pull request merged**